# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/DOMIntersectionObserver.cpp
# Commit: 36115f44f68e
# Full Hash: 36115f44f68e541f6a5e6f44ab93817778d630ed
# Author: Hiroyuki Ikezoe <hikezoe.birchill@mozilla.com>
# Date: 2021-06-14 09:53:07
# Regressor Bug: 1599795
# File Overlap Count: 1
# Description:
#   Bug 1700245 - Replace IsAncestorFrameCrossDoc with IsAncestorFrameCrossDocInProcess in DOMIntersectionObserver::Update. r=emilio
#   
#   Our IntersectionObserver has been working in OOP iframe since bug 1599795.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D117500
# ==============================================================================

diff -r f531f12e5c35 -r 36115f44f68e dom/base/DOMIntersectionObserver.cpp
--- a/dom/base/DOMIntersectionObserver.cpp	Sun Jun 13 19:01:20 2021 +0000
+++ b/dom/base/DOMIntersectionObserver.cpp	Sun Jun 13 22:02:22 2021 +0000
@@ -615,7 +615,8 @@
       // clarification in
       // https://github.com/w3c/IntersectionObserver/issues/456.
       if (rootFrame == targetFrame ||
-          !nsLayoutUtils::IsAncestorFrameCrossDoc(rootFrame, targetFrame)) {
+          !nsLayoutUtils::IsAncestorFrameCrossDocInProcess(rootFrame,
+                                                           targetFrame)) {
         return false;
       }
 
